api/v1beta1/typeoverrides.go

```diff
@@ -58,7 +58,7 @@ type DeploymentV1Spec struct {
 
 	Selector *metav1.LabelSelector `json:"selector,omitempty" protobuf:"bytes,2,opt,name=selector"`
 
-	Template *v14.PodTemplateSpec `json:"template,omitempty" protobuf:"bytes,3,opt,name=template"`
+	Template *DeploymentV1PodTemplateSpec `json:"template,omitempty" protobuf:"bytes,3,opt,name=template"`
```

Comment:
This was required so we can override certain fields of the pod template without providing others.